pkg/pool-weighted/contracts/lib/WeightCompression.sol

```diff
@@ -17,82 +17,42 @@ pragma solidity ^0.7.0;
 import "@balancer-labs/v2-solidity-utils/contracts/math/FixedPoint.sol";
 
 /**
- * @dev Library for compressing and uncompresing numbers by using smaller types.
+ * @dev Library for compressing and decompressing numbers by using smaller types.
  * All values are 18 decimal fixed-point numbers in the [0.0, 1.0] range,
  * so heavier compression (fewer bits) results in fewer decimals.
  */
 library WeightCompression {
-    uint256 private constant _UINT31_MAX = 2**(31) - 1;
-
     using FixedPoint for uint256;
 
-    /**
-     * @dev Convert a 16-bit value to full FixedPoint
-     */
-    function uncompress16(uint256 value) internal pure returns (uint256) {
-        return value.mulUp(FixedPoint.ONE).divUp(type(uint16).max);
-    }
-
-    /**
-     * @dev Compress a FixedPoint value to 16 bits
-     */
-    function compress16(uint256 value) internal pure returns (uint256) {
-        return value.mulUp(type(uint16).max).divUp(FixedPoint.ONE);
+    // If no normalization value is given, assume it is 1.
+    function compress(uint256 value, uint256 bitLength) internal pure returns (uint256) {
+        return compress(value, bitLength, FixedPoint.ONE);
     }
 
-    /**
-     * @dev Convert a 31-bit value to full FixedPoint
-     */
-    function uncompress31(uint256 value) internal pure returns (uint256) {
-        return value.mulUp(FixedPoint.ONE).divUp(_UINT31_MAX);
-    }
-
-    /**
-     * @dev Compress a FixedPoint value to 31 bits
-     */
-    function compress31(uint256 value) internal pure returns (uint256) {
-        return value.mulUp(_UINT31_MAX).divUp(FixedPoint.ONE);
-    }
+    // If a normalization factor is given (e.g., 10_000), normalize against that.
+    function compress(
```

Comment:
The docs for the full `compress` version should explain what it does, and what each argument means. The docs for the overloaded version with reduced arity should then explain that it's just a simpler version where `normalizeTo` is set to FP.ONE.